File: math/number-theory/zeno.py
"""
Copyright https://github.com/microwind
Zeno's Paradox - Dichotomy Paradox
"""

def zeno_paradox(total_distance: float, steps: int) -> None:
    remaining = total_distance
    print("Zeno's Dichotomy Paradox:")
    for i in range(1, steps + 1):
        remaining /= 2
        print(f"Step {i}: remaining distance = {remaining:.6f}")

def main():
    zeno_paradox(1.0, 10)

if __name__ == "__main__":
    main()

File: math/number-theory/zeno.go
// Copyright https://github.com/microwind
// Zeno's Paradox - Dichotomy Paradox

package main

import "fmt"

func zenoParadox(totalDistance float64, steps int) {
	remaining := totalDistance
	fmt.Println("Zeno's Dichotomy Paradox:")
	for i := 1; i <= steps; i++ {
		remaining /= 2
		fmt.Printf("Step %d: remaining distance = %.6f\n", i, remaining)
	}
}

func main() {
	zenoParadox(1.0, 10)
}
